File: java/resistor-color/src/main/java/ResistorColor.java
class ResistorColor {
    int colorCode(String color) {
        if (color.equals("black")) return 0;
        if (color.equals("orange")) return 3;

        return 9;
    }

    String[] colors() {
        return new String[] {
                "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"
        };
    }
}

File: csharp/resistor-color/ResistorColor.cs
using System;
using System.Linq;

public static class ResistorColor
{
    private static readonly string[] COLOR_NAMES = new[] {"black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"};

    public static int ColorCode(string color) => Array.IndexOf(Colors(), color);

    public static string[] Colors() => COLOR_NAMES;
}
